modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java

```diff
@@ -2475,12 +2469,45 @@ private void deleteObsoleteRawSegments() {
                 if (segmentReservedOrLocked(desc.idx))
                     return;
 
-                if (desc.idx < lastCheckpointPtr.index() && duplicateIndices.contains(desc.idx))
-                    segmentAware.addSize(desc.idx, -deleteArchiveFiles(desc.file));
+                if (desc.idx < lastCheckpointPtr.index() && duplicateIndices.contains(desc.idx)) {
+                    long sz = deleteArchiveFiles(desc.file);
+
+                    segmentAware.addSize(desc.idx, -sz);
+                }
             }
         }
     }
 
+    /** {@inheritDoc} */
+    @Override public File compressedSegment(long idx) {
+        return archiveSegment(idx, ZIP_SUFFIX);
```

Comment:
Should return `null` if compression is disabled. Currently it creates a `File` object by path, that never exists.

Reply:
Also `archiveSegment` must return `null` if archive is disabled.